src/kontrol/prove.py

```diff
@@ -436,27 +473,74 @@ def create_kcfg_explore() -> KCFGExplore:
             else:
                 return proof.failure_info
 
-    failure_infos: list[APRFailureInfo | Exception | None]
-    if options.workers > 1:
-        with ProcessPool(ncpus=options.workers) as process_pool:
-            failure_infos = process_pool.map(init_and_run_proof, tests)
-    else:
-        failure_infos = []
-        for test in tests:
-            failure_infos.append(init_and_run_proof(test))
-
-    proofs = [foundry.get_apr_proof(test.id) for test in tests]
-
-    # Reconstruct the proof from the subprocess
-    for proof, failure_info in zip(proofs, failure_infos, strict=True):
-        assert proof.failure_info is None  # Refactor once this fails
-        assert proof.error_info is None
-        if isinstance(failure_info, Exception):
-            proof.error_info = failure_info
-        elif isinstance(failure_info, APRFailureInfo):
-            proof.failure_info = failure_info
-
-    return proofs
+    with Progress(
+        SpinnerColumn(),
+        TimeElapsedColumn(),
+        TextColumn('{task.description}'),
+        TextColumn('{task.fields[status]}'),
+        TextColumn('{task.fields[summary]}'),
+        redirect_stderr=True,
+        redirect_stdout=True,
+    ) as progress:
+
+        failure_infos: list[APRFailureInfo | Exception | None]
+        if options.workers > 1 and len(tests) > 1:
+            done_tests = 0
+            failed_tests = 0
+            passed_tests = 0
+            if not options.hide_status_bar:
+                task = progress.add_task(
+                    f'Multi-proof Mode ({options.workers} workers)',
+                    status='Running',
+                    summary=f'{done_tests}/{len(tests)} completed. {passed_tests} passed. {failed_tests} failed.',
+                )
+
+            def update_status_bar(test_id: str, result: Any) -> None:
+                nonlocal done_tests, failed_tests, passed_tests, progress
+                done_tests += 1
+                proof = foundry.get_apr_proof(test_id)
+                if proof.passed:
+                    passed_tests += 1
+                elif proof.failed:
+                    failed_tests += 1
+                if not options.hide_status_bar:
+                    if progress is not None:
+                        progress.update(
+                            task,
+                            summary=f'{done_tests}/{len(tests)} completed. {passed_tests} passed. {failed_tests} failed.',
+                        )
```

Comment:
You can return early here, and remove a nesting level.
```suggestion
                    if options.hide_status_bar or progress is None:
                        return
                        
                    progress.update(
                        task,
                        summary=f'{done_tests}/{len(tests)} completed. {passed_tests} passed. {failed_tests} failed.',
                    )
```

Reply:
Fixed.